src/character/equipment/equipment.service.spec.ts

```diff
@@ -1,18 +1,96 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { EquipmentService } from './equipment.service';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { EquipmentPosition } from './models/equipment-position.enum';
+import { Equipment } from './models/equipment.entity';
 
 describe('EquipmentService', () => {
-  let service: EquipmentService;
+    let service: EquipmentService;
+    let mockEquipmentRepository = jest.fn();
 
-  beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      providers: [EquipmentService],
-    }).compile();
+    beforeEach(async () => {
+        const module: TestingModule = await Test.createTestingModule({
+            providers: [
+                EquipmentService,
+                {
+                    provide: getRepositoryToken(Equipment),
+                    useValue: mockEquipmentRepository,
+                },
+            ],
+        }).compile();
 
-    service = module.get<EquipmentService>(EquipmentService);
-  });
+        service = module.get<EquipmentService>(EquipmentService);
+    });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
-  });
+    it('should be defined', () => {
+        expect(service).toBeDefined();
+    });
+
+    describe('equipEquipmentPiece', () => {
+        let mockNewPiece;
+        let mockCharacter;
+
+        beforeEach(() => {
+            mockNewPiece = {
+                id: 1,
+                inventory: {
+                    id: 1,
+                },
+                equipable: true,
+                equipmentPosition: EquipmentPosition.CHEST,
+            };
+            mockCharacter = {
+                inventory: {
+                    id: 1,
+                },
+                equipment: {
+                    id: 1,
+                    equipmentPieces: [],
+                },
+            };
+        });
+
+        it('should add the new piece in the equipment', async () => {
+            let equipment = await service.equipEquipmentPiece(mockCharacter, mockNewPiece);
+            let newEquipedPiece = equipment.equipmentPieces.find(e => e.id === mockNewPiece.id);
+            expect(newEquipedPiece).toBeTruthy();
```

Comment:
why not using equipement.equipmentPieces.toContainEqual(mockNewPiece) ?

Reply:
As I'm not using Jest mock I'm not sure if Jest will retreive the same object. Must try